EKF/estimator_interface.cpp

```diff
@@ -248,19 +248,30 @@ void EstimatorInterface::setBaroData(uint64_t time_usec, float data)
 		}
 	}
 
+	// downsample to highest possible sensor rate
+	// by baro data by taking the average of incoming sample
+	_baro_sample_count++;
+	_baro_alt_sum += baro_alt_meter;
+
 	// limit data rate to prevent data being lost
 	if ((time_usec - _time_last_baro) > _min_obs_interval_us) {
 
+		const float baro_alt_avg = _baro_alt_sum / (float)_baro_sample_count;
+
 		baroSample baro_sample_new;
-		baro_sample_new.hgt = data;
-		baro_sample_new.time_us = time_usec - _params.baro_delay_ms * 1000;
+		baro_sample_new.hgt = compensateBaroForDynamicPressure(baro_alt_avg);
```

Comment:
I wanted to stay close to the implementation on the firmware side. We can get the velocity and orientation for compensation also from different source.

Reply:
Yeah but then you're technically fusing future data if you take the middle timestamp.

Reply:
Yeah but you can argue in the same way that you could ending up fusing to old data. It probably just sounds odd that you can fuse future data, but having the delayed fusion horizon allows to fuse the data at the best possible time point.

Reply:
Yeah but fundamentally fusing old data does not violate causality, while future data does.

Reply:
Yes, but having the delayed fusion would allow for non-causal filtering. Do you see a reason why this should not be okay?

Reply:
In practice the causality assumption violated when velocity and quaternion states 100 to 200msec ahead of the fusion time horizon are used for baro compensation is unlikely to be noticeable. However it is easily avoided by performing the correction when data is taken out of the buffer and by using the EKF rather than output observer velocity states so that all data processed is at the same time horizon.

Reply:
Yes. That will be the next PR.